File: rust/src/commands/resolve.rs
use anyhow::Result;
use std::io::Write;

use crate::config::load_config;

pub fn handle_resolve_config_dir() -> Result<()> {
    let profile_name_env = std::env::var("ARC_PROFILE").ok();

    let config = match load_config() {
        Ok(c) => c,
        Err(_) => std::process::exit(1),
    };

    let resolved = profile_name_env
        .as_deref()
        .unwrap_or(&config.active_profile);

    let profile = config.profiles.get(resolved);

    match profile {
        None => std::process::exit(1),
        Some(p) => {
            print!("{}", p.config_dir);
            let _ = std::io::stdout().flush();
            Ok(())
        }
    }
}

File: packages/cli/src/commands/resolve.ts
import { loadConfig } from "../config.js";

export async function handleResolveConfigDir(): Promise<void> {
  const profileName = process.env["ARC_PROFILE"];

  let config;
  try {
    config = loadConfig();
  } catch {
    process.exit(1);
  }

  const resolved = profileName ?? config.activeProfile;
  if (!resolved) {
    process.exit(1);
  }
  const profile = config.profiles[resolved];

  if (!profile) {
    process.exit(1);
  }

  process.stdout.write(profile.configDir);
}
